core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleDecommissionViaCMHandler.java

```diff
@@ -77,22 +83,40 @@ protected Selectable doAccept(HandlerEvent<StopStartDownscaleDecommissionViaCMRe
 
         try {
             Stack stack = stackService.getByIdWithLists(request.getResourceId());
-            Cluster cluster = stack.getCluster();
             ClusterDecomissionService clusterDecomissionService = clusterApiConnectors.getConnector(stack).clusterDecomissionService();
 
             Set<String> hostNames = getHostNamesForPrivateIds(request.getInstanceIdsToDecommission(), stack);
             LOGGER.debug("Attempting to decommission hosts. count={}, hostnames={}", hostNames.size(), hostNames);
 
             Map<String, InstanceMetadataView> hostsToRemove = clusterDecomissionService.collectHostsToRemove(request.getHostGroupName(), hostNames);
+            Map<String, InstanceMetadataView> finalHostsToRemove = hostsToRemove;
             List<String> missingHostsInCm = Collections.emptyList();
             if (hostNames.size() != hostsToRemove.size()) {
                 missingHostsInCm = hostNames.stream()
-                        .filter(h -> !hostsToRemove.containsKey(h))
+                        .filter(h -> !finalHostsToRemove.containsKey(h))
                         .collect(Collectors.toList());
                 LOGGER.info("Found fewer instances in CM to decommission, as compared to initial ask. foundCount={}, initialCount={}, missingHostsInCm={}",
                         hostsToRemove.size(), hostNames.size(), missingHostsInCm);
             }
 
+            List<InstanceMetadataView> instancesWithServicesNotRunning = instanceIdToInstanceMetaDataConverter
+                    .getNotDeletedAndNotZombieInstances(instanceMetaDataService.getAllAvailableInstanceMetadataViewsByStackId(stack.getId()),
+                            request.getHostGroupName(),
+                            request.getRunningInstancesWithServicesNotRunning().stream().map(CloudInstance::getInstanceId).collect(Collectors.toSet()));
+
+            List<InstanceMetadataView> additionalInstancesToDecommission = instancesWithServicesNotRunning
```

Comment:
Can we run this list via the CM API above ? clusterDecommissionService.collectHostsToRemove - or are we relying on the previous step having checked this?
Given we're already callinng CM for this validation - the lists could be combined prior to calling CM.